okhttp/src/main/java/com/squareup/okhttp/ConnectionSpec.java

```diff
@@ -110,11 +114,7 @@ public boolean supportsTlsExtensions() {
 
   /** Applies this spec to {@code sslSocket} for {@code route}. */
   void apply(SSLSocket sslSocket, Route route) {
-    ConnectionSpec specToApply = supportedSpec;
```

Comment:
Caching dropped deliberately, because the enabled set might change? That's fair.

In a follow up I'll be tempted to move the SCSV stuff so we only end up rebuilding this array 1x.